docs: Logcli tutorial and sandbox

Jayclifford345 opened the pull request:
**What this PR does / why we need it**:
This sandbox adds a new LogCLI tutorial based on a fun scenario using logs themed from a logistics centre. 

The PR also modifies the directory structure to accommodate and adds the `detected-fields` option to the reference. 

Here is the sandbox to try: https://killercoda.com/grafana-dev-testing/course/loki/logcli-tutorial

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

trevorwhitney reviewed (approved):
This looks great! I think another section we could add is batch querying lots of logs. This is something `logcli` is great for, that you can't actually do well in another tool like Grafana.

JStickler commented on docs/sources/query/logcli/logcli-tutorial.md:
````diff
@@ -0,0 +1,409 @@
+export LOKI_ADDR=http://localhost:3100
+```
+
+Lets now verify the connection by running the following command:
````
```suggestion
Now let's verify the connection by running the following command:
```

JStickler commented on docs/sources/query/logcli/logcli-tutorial.md:
````diff
@@ -0,0 +1,409 @@
+state
+```
+
+This confirms that LogCLI is connected to the Loki instance and we now know that the logs contain the following labels: `package_size`, `service_name`, and `state`. Lets now run some queries against Loki to better understand our package logistics.
````
```suggestion
This confirms that LogCLI is connected to the Loki instance and we now know that the logs contain the following labels: `package_size`, `service_name`, and `state`. Let's now run some queries against Loki to better understand our package logistics.
```

JStickler commented on docs/sources/query/logcli/logcli-tutorial.md:
```diff
@@ -0,0 +1,409 @@
+
+Instant metric queries are a subset of metric queries that return the value of the metric at a specific point in time. This can be useful for quickly understanding an aggregate state of the stored logs. 
+
+For instance, we can use the following query to get the number of packages being sent from California at the current time:
```
Your introductory text says this query is for "the current time" but it looks like it's for the past five minutes if you don't notice that it's an instant-query.  Maybe call out the change from `query` to `instant-query` in the text to highlight it?

Jayclifford345 replied:
aaah thats a really good point will fix the wording there.

JStickler reviewed (commented):
[Docs team]  I haven't had a chance to walk through the killercoda tutorial yet, I've just read through the text, but wanted to give you the feedback ASAP.

Jayclifford345 commented on docs/sources/query/logcli/logcli-tutorial.md:
````diff
@@ -0,0 +1,409 @@
+     '{service_name="Delivery World"}'
+```
+
+This will write all logs for the `service_name` `Delivery World` in the last 24 hours to the `./inventory` directory. The logs will be split into two files, each containing 12 hours of logs. Note that we do not need to specify `--limit` as this is overridden by the `--parallel-duration` flag. The `--forwards` flag will write the logs in ascending order.
````
Sorry I removed that! good catch

JStickler reviewed (commented):
Had time to run through the sandbox today.  There were a couple of odd results....

Jayclifford345 commented:
Hi @JStickler so sorry i forgot to mention I was at Reinvent fully agree with your original review so still fixing those comments and let me check to see what is happening with the killercoda scenario as I am also seeing that but not locally.

Jayclifford345 commented:
Hi @JStickler, that should be the edit complete and the suggestions added. Just a couple of things to note:
1. Found the route cause of the loki stats returning 0 results. So it appears we only receive stats when Loki needs to return logs from disk or object storage. In our case the demo never does this intime because of the default flush. I have modified the loki config to flush every 5 minutes which isn't best practise but for the purpose of the demo it gives us the result we want. I added a note to let the user know not to do this in production/
2. That should be the wording updated for those sections. Let me know if it sounds better as I agree on return read it didn't make much sense :D. 
3. I sneaked in an extra example for the log_volume as found out you could group the bins by a label which is pretty cool.

Here is the updated sandbox for your review: https://killercoda.com/grafana-dev-testing/course/loki/logcli-tutorial

JStickler reviewed (commented):
[docs team]  Spotted a couple more things on this review, but I think we're getting very close to finished.

Jayclifford345 commented:
Hi @JStickler,
That should be the latest changes in and ready for your review:
https://killercoda.com/grafana-dev-testing/course/loki/logcli-tutorial

JStickler commented on docs/sources/query/logcli/logcli-tutorial.md:
````diff
@@ -0,0 +1,464 @@
+label: timestamp                type: string            cardinality: 20
+```
+
+You can now see why we opted to keep `package_id` in structured metadata and `package_size` as a label. Package ID has a high cardinality and is unique to each log entry, making it a good candidate for structured metadata since we potentially may need to query for it directly. Package size, on the other hand, has a low cardinality and is a good candidate for a label.
````
```suggestion
You can now see why we opted to keep `package_id` in structured metadata and `package_size` as a label. Package ID has a high cardinality and is unique to each log entry, making it a good candidate for structured metadata since we potentially may need to query for it directly. Package size, on the other hand, has a low cardinality, making it a good candidate for a label.
```

JStickler commented on docs/sources/query/logcli/logcli-tutorial.md:
````diff
@@ -0,0 +1,464 @@
+
+This will return a list of all the keys detected in our logs. The output will look similar to the following:
+
+```console
````
Some higher cardinality data for ya:

label: city                                  type: string               cardinality: 15
label: detected_level              type: string               cardinality: 3
label: note                              type: string               cardinality: 7
label: package_id                  type: string               cardinality: 994
label: package_size_extracted              type: string               cardinality: 3
label: package_status           type: string            cardinality: 4
label: package_type             type: string            cardinality: 5
label: receiver_address         type: string            cardinality: 991
label: receiver_name            type: string            cardinality: 100
label: sender_address           type: string            cardinality: 991
label: sender_name              type: string            cardinality: 100
label: state_extracted          type: string            cardinality: 5
label: timestamp                type: string            cardinality: 1000

JStickler reviewed (commented):
[docs team]

Jayclifford345 commented:
Changes ready and waiting for your review @JStickler :)

JStickler reviewed (approved):
[docs team]  LGTM
